src/librustc_trans/cabi_x86_64.rs

```diff
@@ -229,7 +229,11 @@ pub fn compute_abi_info<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>, fty: &mut FnType
 
             if arg.layout.is_aggregate() {
                 let size = arg.layout.size;
-                arg.cast_to(cast_target(cls.as_ref().unwrap(), size))
+                if let Ok(cls) = cls.as_ref() {
+                    if let Some(target) = cast_target(cls, size) {
+                        arg.cast_to(target);
+                    }
```

Comment:
Are the tests now passing because of the `if let Ok(cls)` or the `if let Some(target)`? I feel like instead of silently passing this should fail with a diagnostic error, but I'm not knowledgeable enough about this part fo the language. Otherwise the code seems ready to merge.

@eddyb could you take a look?

Reply:
See https://github.com/rust-lang/rust/issues/38763#issuecomment-357346719 - I think this needs further investigation.